src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/reports/memberdays/MemberDaysReport.java

```diff
@@ -90,96 +75,93 @@ private void addReportDetails(MemberDaysReportData reportData, List<SubmitEvent>
         sortedReportDetails.forEach(d -> reportData.getReportDetails().add(d));
     }
 
-    private void extractValidHearingsFromCurrentCase(List<HearingTypeItem> hearings,
-                                                     CaseData caseData,
-                                                     List<MemberDaysReportDetail> interimReportDetails,
-                                                     ListingData listingData) {
-        for (var hearingItem : hearings) {
-            var dateListedTypeItems = filterHearingsWithHeardStatus(hearingItem);
-            var dateListedItems = filterValidHearingDates(dateListedTypeItems, listingData);
-
-            if (!CollectionUtils.isEmpty(dateListedItems)) {
-                for (var listedItem : dateListedItems) {
-                    var hearingDatePart = getDatePart(listedItem.getValue().getListedDate());
-                    var hearingDate = hearingDatePart != null ?  LocalDate.parse(hearingDatePart) : null;
-                    if (hearingDate != null) {
-                        var reportDetail = new MemberDaysReportDetail();
-                        reportDetail.setSortingHearingDate(hearingDate.toString());
-                        reportDetail.setHearingDate(UtilHelper.formatCurrentDate(hearingDate));
-                        reportDetail.setEmployeeMember(hearingItem.getValue().getHearingEEMember());
-                        reportDetail.setEmployerMember(hearingItem.getValue().getHearingERMember());
-                        reportDetail.setCaseReference(caseData.getEthosCaseReference());
-                        reportDetail.setHearingNumber(hearingItem.getValue().getHearingNumber());
-                        reportDetail.setHearingType(hearingItem.getValue().getHearingType());
-                        reportDetail.setHearingClerk(listedItem.getValue().getHearingClerk());
-                        reportDetail.setHearingDuration(getHearingDuration(listedItem));
-                        reportDetail.setParentHearingId(hearingItem.getId());
-                        interimReportDetails.add(reportDetail);
-                    }
-                }
-            }
+    private void addValidHearingsFromCurrentCase(CaseData caseData,
+                                                 List<MemberDaysReportDetail> reportDetails,
+                                                 ListingData listingData) {
+        var fullPanelHearings = caseData.getHearingCollection().stream()
+            .filter(this::isFullPanelHearing).collect(Collectors.toList());
+
+        for (var hearing : fullPanelHearings) {
```

Comment:
check to see if fullPanelHearings is empty?

Reply:
If fullPanelHearings is empty, it will simply skip the for loop without any errors, so I think that should be fine

Reply:
done